editor/debugger/script_editor_debugger.cpp

```diff
@@ -1408,6 +1432,42 @@ void ScriptEditorDebugger::_bind_methods() {
 	ADD_SIGNAL(MethodInfo("remote_tree_updated"));
 }
 
+void ScriptEditorDebugger::add_debugger_plugin(const Ref<Script> &p_script) {
+	if (!debugger_plugins.has(p_script)) {
+		EditorDebuggerPlugin *plugin = memnew(EditorDebuggerPlugin());
+		plugin->attach_debugger(this);
+		plugin->set_script(p_script);
+		tabs->add_child(plugin);
+		debugger_plugins.insert(p_script, plugin);
+	}
+}
+
+void ScriptEditorDebugger::remove_debugger_plugin(const Ref<Script> &p_script) {
+	if (debugger_plugins.has(p_script)) {
+		debugger_plugins[p_script]->queue_delete();
+		debugger_plugins.erase(p_script);
+	}
+}
+
+void ScriptEditorDebugger::send_message(const String &p_message, const Array &p_args) {
+	MutexLock<Mutex> lock(send_mutex);
```

Comment:
I don't think we need this Mutex. Concurrency at connection level (when supported) is handled by `RemoteDebuggerPeer` itself, and in general, other methods are not thread safe (users should use their own message queue and mutex in GDScript if so desired).